lib/membrane_rtc_engine/endpoints/hls_endpoint.ex

```diff
@@ -63,77 +95,187 @@ if Enum.all?(
-                  description: "Path to directory under which HLS output will be saved",
-                  default: "hls_output"
-                ],
                 owner: [
                   spec: pid(),
                   description: """
                   Pid of parent all notifications will be send to.
-
                   These notifications are:
-                    * `{:playlist_playable, content_type, stream_id, origin}`
+                  * `{:playlist_playable, content_type}`
+                  * `{:cleanup, clean_function}`
                   """
                 ],
-                hls_mode: [
-                  spec: :separate_av | :muxed_av,
-                  default: :separate_av,
+                output_directory: [
+                  spec: Path.t(),
+                  description: "Path to directory under which HLS output will be saved",
+                  default: "hls_output"
+                ],
+                mixer_config: [
+                  spec: MixerConfig.t() | nil,
+                  default: nil,
                   description: """
-                  Defines output mode for `Membrane.HTTPAdaptiveStream.SinkBin`.
-
-                  - `:separate_av` - audio and video tracks will be separated
-                  - `:muxed_av` - audio will be attached to every video track
+                  Audio and video mixer configuration. If you don't want to use compositor pass nil.
                   """
                 ],
-                target_window_duration: [
-                  type: :time,
-                  spec: Membrane.Time.t() | :infinity,
-                  default: Membrane.Time.seconds(20),
+                sink_bin_config: [
+                  spec: SinkBinConfig.t(),
+                  default: SinkBinConfig,
                   description: """
-                  Max duration of stream that will be stored. Segments that are older than window duration will be removed.
                   """
                 ],
-                target_segment_duration: [
-                  type: :time,
-                  spec: Membrane.Time.t(),
-                  default: Membrane.Time.seconds(5),
+                segment_duration: [
+                  spec: SegmentDuration.t(),
+                  default: SegmentDuration.new(Time.seconds(4), Time.seconds(5)),
                   description: """
                   Expected length of each segment. Setting it is not necessary, but
                   may help players achieve better UX.
                   """
                 ],
-                framerate: [
-                  spec: {integer(), integer()} | nil,
-                  description: """
-                  Framerate of input tracks
-                  """,
-                  default: nil
-                ],
-                transcoding_config: [
-                  spec: TranscodingConfig.t(),
-                  default: %TranscodingConfig{},
+                partial_segment_duration: [
```

Comment:
I would merge this with sink_bin_config. You can then use `struct` function for easy HLS struct creation